### PR TITLE
Enable Arcade dependency flow

### PR DESCRIPTION
**Pull request by tmat**
Enables flowing Arcade scripts and SDK automatically to Roslyn.

This means that Roslyn will receive auto-generated and auto-merged PRs that 
1) update global.json to the latest Arcade SDK
2) update scripts in `eng/common` to exactly match the ones in Arcade repo (https://github.com/dotnet/arcade/tree/master/eng/common)
Any changes made to these scripts in Roslyn repo will be discarded.
3) Updates `eng/Versions.Details.xml`, which tracks dependencies across .NET Core stack.

Details can be found at https://github.com/dotnet/arcade/blob/master/Documentation/DependencyFlowOnboarding.md

### CODE REVIEW

**Comment by tmat** [comment]
@jaredpar @markwilkie

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Dependencies>
+  <ProductDependencies>
```
Comment (jaredpar) [inline eng/Version.Details.xml]: Is there a reason we're not taking CoreFX / CoreCLR as dependencies here?

Comment (tmat) [reply eng/Version.Details.xml]: This is just first step to get us plugged in. Once we have this working we can add add other dependencies.

Comment (tmat) [reply eng/Version.Details.xml]: I'd leave that to you :)

**Pull request merged**